src/server/snap.rs

```diff
@@ -239,19 +239,21 @@ impl<R: RaftStoreRouter + 'static> Runnable<Task> for Runner<R> {
             Task::Write(token, mut data) => {
                 SNAP_TASK_COUNTER.with_label_values(&["write"]).inc();
                 match self.files.entry(token) {
-                    Entry::Occupied(mut e) => {
-                        if let Err(err) = data.write_all_to(&mut e.get_mut().0) {
-                            error!(
-                                "failed to write data to snapshot file {} for token {:?}: {:?}",
-                                e.get_mut().0.path(),
-                                token,
-                                err
-                            );
-                            let (_, msg) = e.remove();
-                            let key = SnapKey::from_snap(msg.get_message().get_snapshot()).unwrap();
-                            self.snap_mgr.deregister(&key, &SnapEntry::Receiving);
-                        }
-                    }
+                    Entry::Occupied(mut e) => if let Err(err) =
+                        data.write_all_to(&mut LimiterWriter {
```

Comment:
Is it better to set a rate limiter to the underlying writer instead of hacking it here?

Reply:
I think this fine, becase we might have other io to limit in the future, so implement a common writer is a better choice.

Reply:
@siddontang

Reply:
It's too ugly to wrap every write as a writer.

Reply:
The better way is to use the LimitWriter all the time. But if it is hard to support, I can accept this way now. Or any better idea? @huachaohuang 

Btw, I think we should rename LimiterWriter to LimitWriter